### PR TITLE
feat: event but impl

### PR DESCRIPTION
**Pull request by YoshkiMatryoshki**
Basic implementation for rabbitMq that supports sending simple messages (bytes payload) into predefined queue.

We using default callbacks that should be replaced with correct ones in next updates (maybe(not sure(meeeh)))

Also until entire "notification" feature is done we could use Mock service that will not require any RabbitMQ connection.  New env variable should be empty.

### CODE REVIEW

**Comment by YoshkiMatryoshki** [comment]
https://github.com/tikitko/blog-deploy/pull/2 merge only after this one

File: blog-server-api/src/endpoints/author_subscribe/handler.rs [hunk 1]
```diff
@@ -35,12 +38,17 @@ async fn http_handler(
...
     }
 
-    if author.base.telegram_id.is_none() {
+    if let Some(telegram_id) = author.base.telegram_id {
+        let event = SubscriptionStateChanged {
+            blog_user_id: id,
+            user_telegram_id: telegram_id,
+            new_state: subscribe,
+        };
+        event_bus_service.publish(event).await;
```
Comment (tikitko) [inline blog-server-api/src/endpoints/author_subscribe/handler.rs:47]: Should we await on different queue?

File: blog-server-api/src/endpoints/author_subscribe/request_content.rs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use crate::extensions::Resolve;
 use crate::utils::auth;
-use blog_server_services::traits::author_service::*;
```
Comment (tikitko) [inline blog-server-api/src/endpoints/author_subscribe/request_content.rs:3]: use blog_server_services::traits::author_service::*;
use blog_server_services::traits:: event_bus_service::*;

**Review by tikitko — CHANGES_REQUESTED** [review]

File: blog-server-api/src/extensions.rs [hunk 3]
```diff
@@ -63,13 +66,23 @@ impl Resolve<Arc<Box<dyn EntityPostService>>> for ExtensionsProvider {
...
-pub fn make_extensions(rbatis: RBatis) -> impl ExtensionsProviderType {
+impl Resolve<Arc<Box<dyn EventBusService>>> for ExtensionsProvider {
+    fn resolve(&self) -> Arc<Box<dyn EventBusService>> {
+        self.event_bus_service.clone()
+    }
+}
+
+pub fn make_extensions(
+    rbatis: RBatis,
+    rabbit: Box<dyn EventBusService>,
```
Comment (tikitko) [inline blog-server-api/src/extensions.rs:77]: `Box<dyn EventBusService>` dynamic component -> different lib under the hood -> generic name for param

File: blog-server-api/src/endpoints/author_subscribe/request_content.rs (hunk 2 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/author_subscribe/request_content.rs:3]: Changed

File: blog-server-api/src/endpoints/author_subscribe/handler.rs (hunk 1 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/endpoints/author_subscribe/handler.rs]: yep, now we spawn this one with async move

File: blog-server-api/src/extensions.rs (hunk 3 above)
Comment (YoshkiMatryoshki) [reply blog-server-api/src/extensions.rs]: Name changed

**Review by tikitko — APPROVED** [review]

**Pull request merged**